src/python/test/test_extract_dataset.py

```diff
@@ -0,0 +1,134 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2022 DNAnexus, Inc.
+#
+# This file is part of dx-toolkit (DNAnexus platform client libraries).
+#
+#   Licensed under the Apache License, Version 2.0 (the "License"); you may not
+#   use this file except in compliance with the License. You may obtain a copy
+#   of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#   WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#   License for the specific language governing permissions and limitations
+#   under the License.
+from __future__ import print_function, unicode_literals, division, absolute_import
+
+import unittest
+import tempfile
+import shutil
+import os
+import subprocess
+import pandas as pd
+
+class TestDXExtractDataset(unittest.TestCase):
+    def test_e2e_dataset_ddd(self):
+        dataset_record = "project-G9j1pX00vGPzF2XQ7843k2Jq:record-GBGpKZ80G8j8k4G84P7GP0B5"
+        out_directory = tempfile.mkdtemp()
+        cmd = ["dx", "extract_dataset", dataset_record, "-ddd", "-o", out_directory]
+        subprocess.check_call(cmd)
+        self.end_to_end_ddd(out_directory=out_directory, rec_name = "extract_dataset_test")
+
+    def test_e2e_cohortbrowser_ddd(self):
+        cohort_record = "project-G9j1pX00vGPzF2XQ7843k2Jq:record-GBGq1pQ0vGPf0qYg30Jg7kkG"
```

Comment:
Please try not to hardcode the record but create it programmatically when the tests are run (something very simple, with a couple of fields only required to only test the currently introduced commands).

The created cohort record could be a minimized version of the one you're currently using. This would also make the `truth_output` (below) simpler, more readable.

Reply:
Replaced hard coded record IDs with record names as suggested in [comment](https://github.com/dnanexus/dx-toolkit/pull/786#discussion_r912257101).